### PR TITLE
Support for Output in TTL checks

### PR DESCRIPTION
**Pull request by kinnalru**
Support for passing Output field in consul TTL checks as described here: https://www.consul.io/api/agent/check.html

### CODE REVIEW

**Comment by kinnalru** [comment]
Hi.
@pierresouchay, when you plan to update version of a gem to make "Use correct verbs for Check methods" accessible?

File: lib/diplomat/check.rb [hunk 1]
```diff
@@ -54,28 +54,41 @@ def deregister(check_id)
...
     # @param check_id [String] the unique id of the check
+    # @param status [String] status of the check. Valid values are "passing", "warning", and "critical"
+    # @param output [String] human-readable message will be passed through to the check's Output field
     # @return [Integer] Status code
-    def pass(check_id)
-      ret = @conn.put "/v1/agent/check/pass/#{check_id}"
+    def update_ttl(check_id, status, output = nil)
+      ret = @conn.put do |req|
+        req.url "/v1/agent/check/update/#{check_id}"
+        req.body = JSON.generate('Status' => status, 'Output' => output)
```
Comment (pierresouchay) [inline lib/diplomat/check.rb]: According to the doc, it should use the query parameter "note", not by using JSON payload. Did you test this?

**Comment by pierresouchay** [comment]
@kinnalru I'll produce a new version once we merge this PR, is that OK?

File: lib/diplomat/check.rb (hunk 1 above)
Comment (kinnalru) [reply lib/diplomat/check.rb]: My first production implementation was with "note" query params, but I can't send complex data(json) without URL-encoding. After that I change to generic "/agent/check/update/:check_id" update call and all works well.

There is one issue: Output is silently ignored in "critical" update without any explanation in consul docs. I will open consul issue for this case soon.

**Comment by kinnalru** [comment]
> @kinnalru I'll produce a new version once we merge this PR, is that OK?

ok

File: lib/diplomat/check.rb (hunk 1 above)
Comment (kinnalru) [reply lib/diplomat/check.rb]: warn and pass works as expected in docs

**Comment by pierresouchay** [comment]
@kinnalru Actually, it seems to work with critical as well...
```
#!/usr/bin/env ruby
require 'diplomat'

Diplomat.configure do |config|
  config.url = ENV['CONSUL_HTTP_ADDR'] || "http://localhost:8500"
end
check_id = 'check_ttl'
ttl = Diplomat::Check.register_ttl(check_id, 'check_ttl', 'ttl_check to update', '5m')
puts Diplomat::Check.checks
sleep(3)
puts "Update TTL pass"
Diplomat::Check.pass(check_id, '{"canard": "PASS This is my long output"}')
puts Diplomat::Check.checks
sleep 5
puts "Update TTL warn"
Diplomat::Check.warn(check_id, '{"canard": "WARN This is my long output"}')
puts Diplomat::Check.checks
sleep 5

puts "Update TTL critical"
Diplomat::Check.fail(check_id, '{"canard": "CRITICAL This is my long output"}')
puts Diplomat::Check.checks
sleep 5
```

Outputs:
```
{"check_ttl"=>{"Node"=>"C02X72TUJGH5", "CheckID"=>"check_ttl", "Name"=>"check_ttl", "Status"=>"warning", "Notes"=>"ttl_check to update", "Output"=>"{\"canard\": \"WARN This is my long output\"}", "ServiceID"=>"", "ServiceName"=>"", "ServiceTags"=>[], "Definition"=>{"Interval"=>"", "Timeout"=>"", "DeregisterCriticalServiceAfter"=>""}, "CreateIndex"=>0, "ModifyIndex"=>0}}
Update TTL pass
{"check_ttl"=>{"Node"=>"C02X72TUJGH5", "CheckID"=>"check_ttl", "Name"=>"check_ttl", "Status"=>"passing", "Notes"=>"ttl_check to update", "Output"=>"{\"canard\": \"PASS This is my long output\"}", "ServiceID"=>"", "ServiceName"=>"", "ServiceTags"=>[], "Definition"=>{"Interval"=>"", "Timeout"=>"", "DeregisterCriticalServiceAfter"=>""}, "CreateIndex"=>0, "ModifyIndex"=>0}}
Update TTL warn
{"check_ttl"=>{"Node"=>"C02X72TUJGH5", "CheckID"=>"check_ttl", "Name"=>"check_ttl", "Status"=>"warning", "Notes"=>"ttl_check to update", "Output"=>"{\"canard\": \"WARN This is my long output\"}", "ServiceID"=>"", "ServiceName"=>"", "ServiceTags"=>[], "Definition"=>{"Interval"=>"", "Timeout"=>"", "DeregisterCriticalServiceAfter"=>""}, "CreateIndex"=>0, "ModifyIndex"=>0}}
Update TTL critical
{"check_ttl"=>{"Node"=>"C02X72TUJGH5", "CheckID"=>"check_ttl", "Name"=>"check_ttl", "Status"=>"critical", "Notes"=>"ttl_check to update", "Output"=>"{\"canard\": \"CRITICAL This is my long output\"}", "ServiceID"=>"", "ServiceName"=>"", "ServiceTags"=>[], "Definition"=>{"Interval"=>"", "Timeout"=>"", "DeregisterCriticalServiceAfter"=>""}, "CreateIndex"=>0, "ModifyIndex"=>0}}
```

**Comment by kinnalru** [comment]
> @kinnalru Actually, it seems to work with critical as well...

It's good. So just consul UI not show "critical" output.

**Comment by pierresouchay** [comment]
@kinnalru You should try Consul-UI rich HTML UI for Consul https://github.com/criteo/consul-templaterb/ :-)

The UI source (customizable is here: https://github.com/criteo/consul-templaterb/tree/master/samples/consul-ui )

**Pull request merged**